From 8ed0a0a23a25d6bbb21d163395b39a551e06c207 Mon Sep 17 00:00:00 2001
From: Codex <codex@openai.com>
Date: Mon, 9 Mar 2026 17:44:35 +0000
Subject: [PATCH 3/4] arm64: dts: exynos: starlte: add S6SY761 touchscreen node

---
 .../boot/dts/exynos/exynos9810-starlte.dts    | 24 +++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts b/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts
index 47c724baee..baf1eb3fc6 100644
--- a/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts
+++ b/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts
@@ -160,6 +160,23 @@ s2dos05_buck: buck {
 	};
 };
 
+&hsi2c_10 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	touchscreen@48 {
+		compatible = "samsung,s6sy761";
+		reg = <0x48>;
+		interrupt-parent = <&gpa1>;
+		interrupts = <0 IRQ_TYPE_LEVEL_LOW>;
+		avdd-supply = <&s2dos05_ldo3>;
+		vdd-supply = <&s2dos05_ldo2>;
+		pinctrl-0 = <&ts_int>;
+		pinctrl-names = "default";
+	};
+};
+
 &hsi2c_44 {
 	#address-cells = <1>;
 	#size-cells = <0>;
@@ -317,6 +334,13 @@ sd2_cd: sd2-cd-pins {
 		samsung,pin-pud = <EXYNOS_PIN_PULL_UP>;
 		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV4>;
 	};
+
+	ts_int: ts-int-pins {
+		samsung,pins = "gpa1-0";
+		samsung,pin-function = <EXYNOS_PIN_FUNC_EINT>;
+		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV1>;
+	};
 };
 
 &pinctrl_fsys1 {
-- 
2.43.0

